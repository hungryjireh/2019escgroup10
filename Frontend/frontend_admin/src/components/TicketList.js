import React from "react";
import TicketListItem from "./TicketListItem";
import "../style/TicketList.css";

const TicketList = ({ user, tickets }) => {
  const renderedTickets = tickets.map(ticket => {
    return <TicketListItem user={user} key={ticket.id} ticket={ticket} />;
  });

  return (
    <div className="ticketlist">
      <div className="ticketlist-header">
        <ul className="ticketlist-header-list">
          <li>Subject</li>
          <li>Status</li>
          <li>Requester</li>
          <li>Last Updated</li>
          <li>Group</li>
        </ul>
      </div>
      <div className="ticketlist-body">
        <ul className="ticketlist-body-list">{renderedTickets}</ul>
      </div>
    </div>
  );
};

export default {TicketList, renderedTickets};
