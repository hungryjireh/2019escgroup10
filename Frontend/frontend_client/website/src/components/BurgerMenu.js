import React from "react";
import styled from "styled-components";

import useLockBodyScroll from "./useLockBodyScroll";
import "./BurgerMenuAnimations.css";

const BurgerMenu = styled.div`
  width: 230px;
  height: 100vh;
  position: fixed;
  left: 0;
  padding: 2rem 3rem;
  display: flex;
  flex-direction: column;
  font-family: "Montserrat", sans-serif;
  background-color: white;
  z-index: 5;
`;

export const Darken = styled.div`
  width: 100vw;
  height: 100vh;
  position: fixed;
  background-color: rgb(0, 0, 0);
  opacity: 0.4;
  z-index: 4;
  transition
`;

export const BurgerLink = styled.a`
  font-size: 1.4rem;
  margin-bottom: 3rem;
`;

const BurgerMenuComp = ({ setHasBurgerFalse }) => {
  useLockBodyScroll();
  return (
    <div>
      <BurgerMenu className="slide-in-animation">
        <BurgerLink href="https://beta.acnapi.io">Home</BurgerLink>
        <BurgerLink href="https://beta.acnapi.io/#!/#about_us">About Us</BurgerLink>
        <BurgerLink href="https://beta.acnapi.io/#!/#assets">Our Assets</BurgerLink>
        <BurgerLink href="https://beta.acnapi.io/#!/#case_study">Case Studies</BurgerLink>
        <BurgerLink href="https://beta.acnapi.io/#!/">Contact Us</BurgerLink>
      </BurgerMenu>
      <Darken
        className="darken-animation"
        onClick={() => {
          document
            .querySelector(".slide-in-animation")
            .classList.toggle("slide-out-animation");
          document
            .querySelector(".darken-animation")
            .classList.toggle("lighten-animation");
          setTimeout(() => {
            setHasBurgerFalse();
          }, 400);
        }}
      />
    </div>
  );
};

export default BurgerMenuComp;
