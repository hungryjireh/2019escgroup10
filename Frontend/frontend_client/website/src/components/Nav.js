import React, { useState } from "react";
import ReactDOM from "react-dom";
import styled from "styled-components";

import BurgerMenu from "./BurgerMenu";
import "./Nav.css";

const Nav = styled.nav`
  max-width: 1440px;
  font-size: 1.5rem;
  font-family: "Montserrat", sans-serif;
  background: #000000;
  position: fixed;
  width: 100%;
  z-index: 3;
`;

export const NavList = styled.ul`
  display: flex;
  align-items: center;
`;

export const ListItem = styled.li`
  padding: 1.5rem;
  margin-left: 3rem;
  cursor: pointer
    &:hover > a {
    color: #f9c03f;
  }
`;

export const ItemLink = styled.a`
  color: #fefefe;
  &::selection {
    background-color: #f9c03f;
    color: #2e2e2e;
  }
`;

export const NavBurger = styled.img`
  display: none;

  @media (max-width: 1100px) {
    display: inline;
    margin: 0 2.4rem 0 3rem;
    width: 20px;
    cursor: pointer;
  }
`;

const NavComp = () => {
  const [hasBurger, setHasBurger] = useState(false);

  const handleBurgerClick = () => {
    setHasBurger(true);
  };

  const setHasBurgerFalse = () => {
    setHasBurger(false);
  };

  return (
    <Nav>
      <NavList>
        <ListItem style={{ margin: "0 auto 0 1.6rem" }}>
          <ItemLink data-testid="home-logo" href="https://beta.acnapi.io">
            <img
              className="company-logo"
              src={require("../assets/acnapi-logo-white.png")}
              alt="company logo"
              style={{ width: "83px" }}
            />
          </ItemLink>
        </ListItem>
        <ListItem className="nav-links">
          <ItemLink href="https://beta.acnapi.io/#!/#about_us">About Us</ItemLink>
        </ListItem>
        <ListItem className="nav-links">
          <ItemLink href="https://beta.acnapi.io/#!/#assets">Our Assets</ItemLink>
        </ListItem>
        <ListItem className="nav-links">
          <ItemLink href="https://beta.acnapi.io/#!/#case_study">Case Studies</ItemLink>
        </ListItem>
        <ListItem className="nav-links" style={{ margin: "0 1.6rem 0 3rem" }}>
          <ItemLink href="https://beta.acnapi.io/#!/">Contact Us</ItemLink>
        </ListItem>
        <NavBurger
          onClick={handleBurgerClick}
          src={require("../assets/burger-white.svg")}
          alt="nav burger"
        />
      </NavList>
      {hasBurger &&
        ReactDOM.createPortal(
          <BurgerMenu setHasBurgerFalse={setHasBurgerFalse} />,
          document.querySelector("#burger")
        )}
    </Nav>
  );
};

export default NavComp;
