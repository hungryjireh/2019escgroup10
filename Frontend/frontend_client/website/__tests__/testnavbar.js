import React from "react";
import { render, cleanup, fireEvent, waitForElement, getByLabelText, getByText, getByTestId } from 'react-testing-library';
import {ItemLink, NavBurger, NavList, ListItem} from "../components/Nav";
import Nav from "../components/Nav";

afterEach(cleanup);

it("ItemLink renders", () => {
    const { asFragment } = render(<ItemLink />);
    expect(asFragment()).toMatchSnapshot();
});

it("NavBurger renders", () => {
    const { asFragment } = render(<NavBurger />);
    expect(asFragment()).toMatchSnapshot();
});

it("ListItem renders", () => {
    const { asFragment } = render(<ListItem />);
    expect(asFragment()).toMatchSnapshot();
});

it("NavList renders", () => {
    const { asFragment } = render(<NavList />);
    expect(asFragment()).toMatchSnapshot();
});

test('assert the image link to be home link', () => {
    const { getByTestId } = render(<Nav />);
    const allAgent = getByTestId('home-logo');
    expect(getByTestId('home-logo').hasAttribute('href'));
    expect(allAgent.getAttribute('href')).toBe('https://beta.acnapi.io');
})

test('assert the first item link to be About Us link', () => {
    const { getByText, getByAltText } = render(<Nav />);
    const allAgent = document.querySelector('.nav-links').innerHTML;
    expect(allAgent).toContain('https://beta.acnapi.io/#!/#about_us');
})

test('assert the second item link to be Assets link', () => {
    const { getByText, getByAltText } = render(<Nav />);
    const allAgent = document.querySelectorAll('.nav-links')[1].innerHTML;
    expect(allAgent).toContain('https://beta.acnapi.io/#!/#assets');
})

test('assert the third item link to be Case Studies link', () => {
    const { getByText, getByAltText } = render(<Nav />);
    const allAgent = document.querySelectorAll('.nav-links')[2].innerHTML;
    expect(allAgent).toContain('https://beta.acnapi.io/#!/#case_study');
})

test('Navbar elements all present', () => {
    const { getByText, getByAltText } = render(<Nav />);
    expect(document.querySelectorAll(".nav-links").length).toBe(4);
    expect(document.querySelectorAll(".company-logo").length).toBe(1);
    expect(getByAltText("nav burger"));
})

test('calls "onClick" prop on button click of ItemLink with text', () => {
    const onClick = jest.fn();
    const { getByText } = render(<ItemLink onClick={onClick}>About Us</ItemLink>);

    fireEvent.click(getByText(/About Us/i));
    expect(onClick).toHaveBeenCalled();
});

test('calls "onClick" prop on button click of ItemLink with image', () => {
    const onClick = jest.fn();
    const { getByAltText } = render(<ItemLink><img onClick={onClick} alt="company logo"></img></ItemLink>);

    fireEvent.click(getByAltText(/company logo/i));
    expect(onClick).toHaveBeenCalled();
});

test('calls "onClick" prop on button click of NavBurger', () => {
    const onClick = jest.fn();
    const { getByAltText } = render(<NavBurger onClick={onClick} alt="nav burger" />);

    fireEvent.click(getByAltText(/nav burger/i));
    expect(onClick).toHaveBeenCalled();
});