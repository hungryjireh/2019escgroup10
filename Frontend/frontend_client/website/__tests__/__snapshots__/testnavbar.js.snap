// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemLink renders 1`] = `
<DocumentFragment>
  <a
    class="sc-bZQynM kyAjQX"
  />
</DocumentFragment>
`;

exports[`ListItem renders 1`] = `
<DocumentFragment>
  <li
    class="sc-EHOje gQjhhH"
  />
</DocumentFragment>
`;

exports[`NavBurger renders 1`] = `
<DocumentFragment>
  <img
    class="sc-gzVnrw cgODWv"
  />
</DocumentFragment>
`;

exports[`NavList renders 1`] = `
<DocumentFragment>
  <ul
    class="sc-ifAKCX dCoIjQ"
  />
</DocumentFragment>
`;

exports[`renders the Nav 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-bxivhb bBitEP"
      />
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-bxivhb bBitEP"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
